src/Fhir.Anonymizer.Core/PartitionedExecution/FhirPartitionedExecutor.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Fhir.Anonymizer.Core.PartitionedExecution;
+
+namespace Fhir.Anonymizer.Core
+{
+    public class FhirPartitionedExecutor
+    {
+        public IFhirDataReader RawDataReader { set; get; }
+
+        public IFhirDataConsumer AnonymizedDataConsumer { set; get; }
+
+        public Func<string, string> AnonymizerFunction { set; get; }
```

Comment:
There's nothing specific to anonymization here. How about making it more generic by using "transform" instead of "anonymize" in this file?